### PR TITLE
[ARGO-801] - New web api python probe

### PR DESCRIPTION
**Pull request by themiszamani**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,28 @@
 # argo-probes
+
+## Web API 
+
+This is a probe for checking AR and status reports are properly working. 
+It checks if there are available AR and status data for a selected day. 
+
+The usage of the script is:
+```sh
+$ usage: api.py [-h] -H HOSTNAME -tenant TENANT -rtype RTYPE -token TOKEN
```
Comment (vrdel) [inline README.md]: web-api.py [-h] ...

File: modules/web-api.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        raise SystemExit(1)
+    elif NAGIOS_RESULT == 2:
+        print 'CRITICAL - Problem with  %s reports for tenant %s return results  ' % (arguments.rtype, arguments.tenant)
+        print 'Description:'+Description
+        debugValues(arguments)
+        raise SystemExit(2)
+
+if __name__ == "__main__":
+    main()
+                                                                                                                                                                                                                              252,1         B
```
Comment (vrdel) [inline modules/web-api.py]: Where did this come from?

File: modules/web-api.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-H', dest='hostname', required=True, type=str, help='hostname')
+    parser.add_argument('-tenant', dest='tenant', required=True, type=str, default='EGI', help='tenant to check. Default EGI')
+    parser.add_argument('-rtype', dest='rtype', required=True, type=str, default='ar', help='status or ar (default ar)')
+    parser.add_argument('-token', dest='token', required=True, type=str, default='test_token', help='authentication token')
+    parser.add_argument('-day', dest='day', required=False, type=int, default=1, help='days to check (ex. 1 for yesterday, 2 for days ago) default yesterday')
+    parser.add_argument('-t', dest='timeout', required=False, type=int, default=180)
+    parser.add_argument('-d', dest='debug', required=False, type=str, default='off')
```
Comment (vrdel) [inline modules/web-api.py]: You can have:
``
parser.add_argument('-d', dest='debug', required=False, action='store_true', default=False)
``
for argument to act as boolean.

File: modules/web-api.py [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+        Args:
+            tenant: the tenant to test
+            hostname: we use the hostname to check for devel, or production
+            reportName: the name of the report to check
+    """
+    #reports we dont use in production
+    EGI_prod = ['Cloud', 'Critical-Fedcloud', 'Fedcloud']
+    EGI_devel = ['Cloud', 'Critical-Fedcloud', 'Fedcloud']
+
+    if (tenant == 'EGI' and hostname == 'api.argo.grnet.gr'):
```
Comment (vrdel) [inline modules/web-api.py]: You should **not** have hardcoded stuff in probe (nor tenant nor hostaname). I think that checkUnusedReports() should be removed and that you should simply introduce 

``
--unused-reports Cloud Critical-Fedcloud Fedcloud
``

So if for given tenant and hostname, ``--unused-reports`` are specified, probe logic should be to skip those reports. ``--unused--reports`` should be optional argument.

File: modules/web-api.py (hunk 2 above)
Comment (themiszamani) [inline modules/web-api.py]: Dont know .. i erased it.

File: README.md (hunk 1 above)
Comment (themiszamani) [inline README.md]: done

File: modules/web-api.py (hunk 3 above)
Comment (themiszamani) [inline modules/web-api.py]: its i parse with the on and off values.

File: modules/web-api.py (hunk 4 above)
Comment (themiszamani) [inline modules/web-api.py]: i added a new jira ticket so as to update the probe this week.. 
https://jira.argo.grnet.gr/browse/ARGO-822

File: modules/web-api.py (hunk 3 above)
Comment (vrdel) [inline modules/web-api.py]: ``-d`` should be ``-v``, meaning verbose argument and it should be simple "flag" argument without any additional arguments. Check: https://nagios-plugins.org/doc/guidelines.html

File: modules/web-api.py [hunk 5]
```diff
@@ -209,6 +204,7 @@ def main():
     parser.add_argument('-tenant', dest='tenant', required=True, type=str, default='EGI', help='tenant to check. Default EGI')
     parser.add_argument('-rtype', dest='rtype', required=True, type=str, default='ar', help='status or ar (default ar)')
     parser.add_argument('-token', dest='token', required=True, type=str, default='test_token', help='authentication token')
+    parser.add_argument('-unused-reports', dest='unusedreports', required=False, type=str,help='Add unused reports from API as a string separated by comma')
```
Comment (vrdel) [inline modules/web-api.py]: Add ``nargs='+'`` so argument will swallow everything specified to it into ``args.unusedreports`` which will be of type list. That way you don't need to imply comma separated list of reports enclosed in quotes. You can just simply say: ``--unused-reports Report Report2``. List should be also easier to handle further in code.
Also, multicharacter argument are longoptions (see getopt(1)) and those should start with ``'--'``

**Comment by vrdel** [comment]
+1

**Pull request merged**